File: back_fastapi/app/schemas/order.py
from decimal import Decimal
from typing import List

from pydantic import BaseModel

from app.models.order import OrderModeEnum, OrderStatusEnum


class OrderItemCreate(BaseModel):
  product_id: int
  quantity: int


class OrderCreate(BaseModel):
  mode: OrderModeEnum
  items: List[OrderItemCreate]
  address: str | None = None
  comment: str | None = None


class OrderItemOut(BaseModel):
  id: int
  product_id: int
  product_name: str
  product_price: Decimal
  quantity: int

  class Config:
    from_attributes = True


class OrderOut(BaseModel):
  id: int
  mode: OrderModeEnum
  status: OrderStatusEnum
  total_price: Decimal
  address: str | None
  comment: str | None
  items: List[OrderItemOut]

  class Config:
    from_attributes = True

File: front_nextjs/libs/api/order.ts
import { apiFetch } from "./apiFetch";

export type OrderMode = "delivery" | "restaurant";

export type OrderItemCreate = {
  product_id: number;
  quantity: number;
};

export type OrderCreateRequest = {
  mode: OrderMode;
  items: OrderItemCreate[];
  address?: string | null;
  comment?: string | null;
};

export type OrderItemOut = {
  id: number;
  product_id: number;
  product_name: string;
  product_price: string;
  quantity: number;
};

export type OrderOut = {
  id: number;
  mode: OrderMode;
  status: "pending" | "confirmed" | "completed" | "cancelled";
  total_price: string;
  address?: string | null;
  comment?: string | null;
  items: OrderItemOut[];
};

export async function createOrder(
  payload: OrderCreateRequest,
): Promise<OrderOut> {
  const res = await apiFetch("/orders", {
    method: "POST",
    body: JSON.stringify(payload),
  });
  return res.json();
}

export async function getOrders(): Promise<OrderOut[]> {
  const res = await apiFetch("/orders", { method: "GET" });
  return res.json();
}
